frontend/src/components/Playback/SubtitleTrack.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+    <div
+        class="subtitle-track"
+        :label="playerElement.currentExternalSubtitleTrack?.label"
+        :srclang="playerElement.currentExternalSubtitleTrack?.srcLang"
+        :src="playerElement.currentExternalSubtitleTrack?.src" >
+        <span
+          class="subtitle-track-text"
+          :style="subtitleStyle">
+          <JSafeHtml
+          v-if="currentSubtitle !== undefined"
+          :html="currentSubtitle.text" />
+          {{ previewText }}
+        </span>
+    </div>
+</template>
+
+<script setup>
+import { computed, defineProps } from 'vue';
+import { clientSettings } from '@/store/client-settings';
+import { mediaControls } from '@/store';
+import { playerElement } from '@/store/player-element';
+import { isNil } from '@/utils/validation';
+
+defineProps({
+  previewText: String
+});
```

Comment:
See how we define the props in other places. This is using Vue built-in types, but if we use TypeScript interfaces, we could make it optional much easily:

```suggestion
defineProps<{
  previewText?: string;
}>();
```

Which is the equivalent with Vue types to something like this (might not work since I'm typing this by heart and it's been a while since I don't use this types):

```suggestion
defineProps({
  previewText: {
     type: String,
     default: undefined
});
```

Yeah, the Vue types should be deprecated altogether in a future Vue 4 because it's all super confusing :P